frontend/src/components/problem/editor/OptionsPanel.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useHistory } from 'react-router-dom';
+import { LowMarginMediumText, SmallHeaderText, Text } from '../../core/Text';
+import ToggleButton from '../../core/ToggleButton';
+import { Difficulty } from '../../../api/Difficulty';
+import {
+  GrayTextButton, PrimaryButton, SmallDifficultyButton, TextButton,
+} from '../../core/Button';
+import { generateRandomId, validIdentifier } from '../../../util/Utility';
+import ProblemTags from '../ProblemTags';
+import { TextInput } from '../../core/Input';
+import PrimarySelect from '../../core/Select';
+import {
+  deleteProblem, ProblemIOType, problemIOTypeToString, ProblemTag, Problem,
+} from '../../../api/Problem';
+import { SettingsContainer } from '../../core/Container';
+import { HoverTooltip } from '../../core/HoverTooltip';
+import { useAppSelector, useMousePosition, useProblemEditable } from '../../../util/Hook';
+import { AccountRole } from '../../../api/Account';
+import { InlineErrorIcon } from '../../core/Icon';
+
+const SidebarContent = styled.div`
+  text-align: left;
+  padding: 20px;
+  flex: 4;
+`;
+
+type ShowProps = {
+  show: boolean,
+};
+
+const VerifiedContainer = styled.div<ShowProps>`
+  display: ${({ show }) => (show ? 'inline-block' : 'none')};
+  text-align: left;
+  margin-top: 0.5rem;
+`;
+
+const VerifiedText = styled(Text)`
+  display: inline-block;
+  margin: 0 0 0 0.75rem;
+  font-size: ${({ theme }) => theme.fontSize.subtitleXMediumLarge};
+`;
+
+const InputTypeContainer = styled.div`
+  margin-bottom: 5px;
+`;
+
+const CancelTextButton = styled(TextButton)`
+  margin-left: 2.5px;
+  color: ${({ theme }) => theme.colors.gray};
+`;
+
+const DeleteButton = styled(PrimaryButton)`
+  margin: 0 0 1rem 0;
+  color: ${({ theme }) => theme.colors.red2};
+  background: ${({ theme }) => theme.colors.white};
+`;
+
+type OptionsPanelProps = {
+  newProblem: Problem,
+  setNewProblem: (problem: Problem) => void,
+  editMode: boolean,
+  setLoading: (loading: boolean) => void,
+  setError: (error: string) => void,
+};
+
+function OptionsPanel(props: OptionsPanelProps) {
+  const {
+    newProblem, setNewProblem, editMode, setLoading, setError,
+  } = props;
+
+  const history = useHistory();
+  const [hoverVisible, setHoverVisible] = useState<boolean>(false);
+
+  const { account, firebaseUser, token } = useAppSelector((state) => state.account);
+  const mousePosition = useMousePosition();
```

Comment:
Change this to take in the `true` attribute; this occurs when the user enters a parameter name that is likely invalid, a little warning pops up (see screenshot below).

**Variable Name is Likely Invalid Popup**
<img width="1274" alt="Variable Name is Likely Invalid Popup" src="https://user-images.githubusercontent.com/7987279/121826101-e2e21880-cc6a-11eb-8753-b31985965574.png">